marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/hub-es.sjs

```diff
@@ -121,8 +121,23 @@ function getRefEntityIdentifiers(entityIRI, entityInfo, prop) {
   return refEntityId;
 }
 
+function getEntityServiceTitle(iri) {
+  if (!(iri instanceof sem.iri)) {
+    iri = sem.iri(iri);
+  }
+  const triple = fn.head(cts.triples(iri,
```

Comment:
Thoughts on parsing this out instead of doing a query? I'm just thinking that for mapping 1 million docs, this means 1 million additional triple lookups. 

I've also tried to start a naming convention in entity-lib.sjs where "find" indicates that a function will do a query, and "get" indicates that a function will not do a query. I've found this to be helpful on previous projects, though of course this is just one of dozens of DH libraries. But I do think it's a helpful way so readers can quickly detect where queries occur when analyzing code for performance.

Reply:
@rjrudin I'm trying to recall, did we add validation to the entity title to ensure we don't have spaces and other things that potentially require knowing the right decoding rules to get the value from the URI? If we have that, I don't have an issue with parsing it from the IRI. 

I can also make some adjustments to limit the triples query to the models and cache the value for the duration of a batch/transaction.

Reply:
We do have validation in the new model endpoints in 5.3, but of course that won't impact existing entity models. Though since we'll have docs telling people how to upgrade their entity models for HC and 5.3, we could tell them to ensure their titles conform to the validation rules in 5.3.

entity-lib.sjs has a function too for doing the parsing - getEntityTypeIdParts .